app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/OwnedListFragment.java

```diff
@@ -200,11 +208,42 @@ private void showFilterPopup(View view) {
         int height = layout.getMeasuredHeight();
         PopupWindow popup = new PopupWindow(layout, width, height,true);
 
+        SwitchMaterial availableStatusSwitch = layout.findViewById(R.id.filter_menu_available);
+        SwitchMaterial requestedStatusSwitch = layout.findViewById(R.id.filter_menu_requested);
+        SwitchMaterial acceptedStatusSwitch = layout.findViewById(R.id.filter_menu_accepted);
+        SwitchMaterial borrowedStatusSwitch = layout.findViewById(R.id.filter_menu_borrowed);
+        availableStatusSwitch.setOnClickListener(this::onFilter);
+        requestedStatusSwitch.setOnClickListener(this::onFilter);
+        acceptedStatusSwitch.setOnClickListener(this::onFilter);
+        borrowedStatusSwitch.setOnClickListener(this::onFilter);
```

Comment:
better design is to let the parent view handle the click... tried a lot of options but couldn't make it work